infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/AsyncStorageProvisioner.java

```diff
@@ -0,0 +1,370 @@
+    List<SshPairImpl> sshPairs = getOrCreateSshPairs(identity, osEnv);
+    if (sshPairs == null) {
+      return null;
+    }
+    SshPair sshPair = sshPairs.get(0);
+    Map<String, String> sshConfigData =
+        ImmutableMap.of(AUTHORIZED_KEYS, sshPair.getPublicKey() + "\n");
+    return new ConfigMapBuilder()
+        .withNewMetadata()
+        .withName(configMapName)
+        .withNamespace(namespace)
+        .endMetadata()
+        .withData(sshConfigData)
+        .build();
+  }
+
+  /**
+   * Create storage Pod with container with mounted volume for storing project source backups, SSH
+   * key and exposed port for rsync connection
+   */
+  private Pod createStoragePod(String namespace, String configMap) {
+    String containerName = Names.generateName(ASYNC_STORAGE);
+
+    Volume storageVolume =
+        new VolumeBuilder()
+            .withName(STORAGE_VOLUME)
+            .withPersistentVolumeClaim(
+                new PersistentVolumeClaimVolumeSourceBuilder()
+                    .withClaimName(configuredPVCName)
+                    .withReadOnly(false)
+                    .build())
+            .build();
+
+    Volume sshKeyVolume =
+        new VolumeBuilder()
+            .withName(CONFIG_MAP_VOLUME_NAME)
+            .withConfigMap(
+                new ConfigMapVolumeSourceBuilder()
+                    .withName(configMap)
+                    .withDefaultMode(0600)
+                    .build())
+            .build();
+
+    VolumeMount storageVolumeMount =
+        new VolumeMountBuilder()
+            .withMountPath(ASYNC_STORAGE_DATA_PATH)
+            .withName(STORAGE_VOLUME)
+            .withReadOnly(false)
+            .build();
+
+    VolumeMount sshVolumeMount =
+        new VolumeMountBuilder()
+            .withMountPath(SSH_KEY_PATH)
+            .withSubPath(AUTHORIZED_KEYS)
+            .withName(CONFIG_MAP_VOLUME_NAME)
+            .withReadOnly(true)
+            .build();
+
+    Container container =
+        new ContainerBuilder()
+            .withName(containerName)
+            .withImage(storageImage)
+            .withNewResources()
+            .addToLimits("memory", new Quantity("512Mi"))
+            .addToRequests("memory", new Quantity("256Mi"))
+            .endResources()
+            .withPorts(
+                new ContainerPortBuilder()
+                    .withContainerPort(SERVICE_PORT)
+                    .withProtocol("TCP")
+                    .build())
+            .withVolumeMounts(storageVolumeMount, sshVolumeMount)
+            .build();
+
+    PodSpecBuilder podSpecBuilder = new PodSpecBuilder();
+    PodSpec podSpec =
+        podSpecBuilder.withContainers(container).withVolumes(storageVolume, sshKeyVolume).build();
+
+    return new PodBuilder()
+        .withApiVersion("v1")
+        .withKind("Pod")
+        .withNewMetadata()
+        .withName(ASYNC_STORAGE)
+        .withNamespace(namespace)
+        .withLabels(singletonMap("app", ASYNC_STORAGE))
```

Comment:
As @mshaposhnik mentions above, try to use utility methods that would also apply standard labels and annotations to the pod. Here, we could probably use `KubernetesDeployments`.

Reply:
https://github.com/eclipse/che/issues/17400